timesketch/lib/analyzers/interface.py

```diff
@@ -103,14 +105,33 @@ def __init__(self, event, datastore, sketch=None):
             raise KeyError('Malformed event: {0:s}'.format(e))
 
     def _update(self, event):
-        """Update Event in Elasticsearch.
+        """Update event attributes to add.
 
         Args:
             event: Dictionary with new or updated values.
         """
+        self.updated_event.update(event)
+
+    def commit(self, event_dict=None):
+        """Commit an event to Elasticsearch.
+
+        Args:
+            event_dict: (optional) Dictionary with updated event attributes.
```

Comment:
maybe add that if not supplied it will default to self.updated_event

Reply:
Done.